File: python3/factorial.py

num = int(input("Enter a number: "))
result = 1

for var in range(1, num+1):
    result *= var
print("!",num," es ",result)

#pide numero hasta introduce 0 termina, salida imprirlos y sus media
cero = True
lista = []
suma = 0

while cero:
    n = int(input("Enter a number: "))
    if n == 0:
        cero = False
    else:
        lista.append(n)
        suma += n
        
print("numeros introducidos son %s , y la media es %s"%(lista,(suma/len(lista))))

File: c/Factorial.cpp
#include <iostream>
using namespace std;

int main()
{
    int x;
    int cont = 1, result = 1;

    cout << "Enter a number: " << endl;
    cin >> x;

    while (cont <= x)
    {
        result *= cont;
        cont++;
    }

    cout << "Factorial of !" << x << " is " << result << endl;
}